Earthfile

```diff
@@ -12,290 +12,456 @@
-  COPY +spaceros-artifacts/${SPACEROS_DIR}/exact.repos exact.repos
-  SAVE ARTIFACT exact.repos AS LOCAL exact.repos
+  # Rosdep updates
+  RUN apt-get update && apt-get install -y python3-rosdep \
+      && rosdep init \
+      && rosdep update
 
-rosdep:
-  FROM +sources
-  COPY src src/
+  # Copy Repos file
+  COPY +sources/src ./src
+  COPY excluded-pkgs.txt excluded-deps.txt ./
 
   # Install system package dependencies using rosdep
-  RUN sudo rosdep init && rosdep update
-  RUN --mount=type=cache,mode=0777,target=/var/cache/apt,sharing=locked,id=cache_apt_cache \
-      --mount=type=cache,mode=0777,target=/var/lib/apt,sharing=locked,id=lib_apt_cache \
-      sudo apt-get update && \
-      rosdep update && \
-      rosdep install -s -y \
+  RUN rosdep install -y \
         --from-paths src --ignore-src \
+        --simulate \
         --rosdistro ${ROS_DISTRO} \
         # `urdfdom_headers` is cloned from source, however rosdep can't find it.
         # It is because package.xml manifest is missing. See: https://github.com/ros/urdfdom_headers
         # Additionally, IKOS must be excluded as per: https://github.com/space-ros/docker/issues/99
-        --skip-keys "$(tr '\n' ' ' < 'excluded-pkgs.txt') urdfdom_headers ikos" > rosdep-commands.sh && \
-      chmod u+x rosdep-commands.sh && \
-      ./rosdep-commands.sh
-        
-  RUN rm excluded-pkgs.txt
-
-  RUN --mount=type=cache,mode=0777,target=/var/cache/apt,sharing=locked,id=cache_apt_cache \
-      --mount=type=cache,mode=0777,target=/var/lib/apt,sharing=locked,id=lib_apt_cache \
-      sudo apt-get update && \
-      sudo apt-get install --yes \
-        gcc \
-        g++ \
-        cmake \
-        libgmp-dev \
-        libboost-dev \
-        libboost-filesystem-dev \
-        libboost-thread-dev \
-        libboost-test-dev \
-        libsqlite3-dev \
-        libtbb-dev \
-        libz-dev \
-        libedit-dev \
-        python3 \
-        python3-pip \
-        python3-venv \
-        llvm-14 \
-        llvm-14-dev \
-        llvm-14-tools \
-        clang-14
-
-  WORKDIR ${SPACEROS_DIR}
-  RUN git clone --branch v3.4 --depth 1 https://github.com/NASA-SW-VnV/ikos.git
-  WORKDIR ${SPACEROS_DIR}/ikos
-  RUN mkdir build
-  WORKDIR ${SPACEROS_DIR}/ikos/build
-  RUN cmake \
-        -DCMAKE_INSTALL_PREFIX="/opt/ikos" \
-        -DCMAKE_BUILD_TYPE="Debug" \
-        -DLLVM_CONFIG_EXECUTABLE="/usr/lib/llvm-14/bin/llvm-config" \
-        ..
-  RUN make -j`nproc`
-  RUN sudo make install
-  ENV PATH="/opt/ikos/bin/:$PATH"
-  WORKDIR ${SPACEROS_DIR}
-  RUN sudo rm -rf ikos/
-
+        --skip-keys "$(tr '\n' ' ' < 'excluded-pkgs.txt') urdfdom_headers ikos" > rosdeps.txt
+
+  # Process rosdeps.txt to a shell script
+  RUN touch rosdeps.sh \
+        && echo "#!/bin/bash" > rosdeps.sh \
+        && echo "apt-get update" >> rosdeps.sh \
+        && echo "apt-get install -y \\" >> rosdeps.sh \
+        && grep -v -F -f excluded-deps.txt rosdeps.txt | sed 's/^/  /' >> rosdeps.sh \
```

Comment:
Why can't this textual content be a file that is copied in?

Reply:
ah, I see, because it depends on `rosdeps.txt` which is generated at build-time. not sure I like this workflow but I answered my own question